Source/Editor/DualityEditor/DualityEditorApp.cs

```diff
@@ -847,18 +850,44 @@ public static void ReadPluginSourceCodeContentData(out string rootNamespace, out
 				}
 			}
 		}
-		public static void InitPluginSourceCode()
+
+		/// <summary>
+		/// Initializes the solution unless there already is an sln file.
+		/// </summary>
+		/// <returns>The path to the solution file</returns>
+		public static string InitPluginSourceCode()
 		{
-			// Create solution file if not existing yet
-			if (!File.Exists(EditorHelper.SourceCodeSolutionFile))
+			// Check if the solution file has to be created
+			string solution = FindSolutionFile();
+			if (solution == null)
 			{
-				using (MemoryStream gamePluginStream = new MemoryStream(Properties.GeneralRes.GamePluginTemplate))
-				using (ZipArchive gamePluginZip = new ZipArchive(gamePluginStream))
-				{
-					gamePluginZip.ExtractAll(EditorHelper.SourceCodeDirectory, false);
-				}
+				CreateDefaultSolution();
+				solution = FindSolutionFile();
+				// Shouldnt happen since the Properties.GeneralRes.GamePluginTemplate should always contain a solution file.
+				if (solution == null)
+					throw new FileNotFoundException("Could not find a solution file. The Properties.GeneralRes.GamePluginTemplate might have been changed and no longer contains a sln file.");
 			}
-			
+			return solution;
+		}
+
+		private static string FindSolutionFile()
```

Comment:
Suggestion: Consider merging the contents of this method with the updated `SolutionFilePath` property, since it won't be used anywhere else after the property change and is so small anyway. Otherwise, please move alongside the property into `EditorHelper` to keep them as close location-wise, as they're content-wise.